functions.R

```diff
@@ -881,3 +891,150 @@ plotter <- function(df, variable = NULL, show_labels = TRUE){
   }
 }
 
+# taking factors with one level in osduhs survey data and imputing 'NO' onto it
+relevel_factor_one_lfs <- function(dat_var) {
+  dat_var <- as.character(dat_var)
+  dat_var[is.na(dat_var)] <- 'NO'
+  return(dat_var)
+}
+
+# restructure data types
+restructure_data_types <- function(temp) {
+  for(i in 1:ncol(temp)) {
+    if(grepl('factor', class(temp[, i]))) {
+      temp[, i] <- as.character(temp[, i])
+    } else {
+      temp[, i] <- as.numeric(temp[, i])
+    }
+  }
+  return(temp)
+}
+
+clean_lfs <- function(temp_clean) {
+  
+  # characters back to factors, and everything else numeric
+  temp_clean <- restructure_data_types(temp_clean)
+  
+  # remove extra columns or columns with too many NAs
+  temp_clean$current_student_status_and_type_of_school.1 <- NULL
+  
+  # if the level of a factor is 1, then that factor only has "yes" coded and should replace NA with "NO"
```

Comment:
@benmbrew This is a good example of a place where it'd be easier to use dplyr. Functionally, the code would do the same thing, but rather than having to write `temp_clean$` twice on every line, you could do something like:

```
temp_clean <- temp_clean %>%
mutate(columna = relevel_factor_one_lfs(columna),
            columnb = relevel_factor_one_lfs(columnb))
(etc)
```
No need to change this.

Reply:
Oh I really like this one. I didn't know mutate could do that.